File: Squares_and_Cubes_1619B.py
t = int(input())
while(t):
    n = int(input())
    soc = [1]
    for i in range(n):
        if i ** 2 <= n:
            soc.append(i ** 2)
        else:
            break
        if i ** 3 <= n:
            soc.append(i ** 3)
    soc= set(soc)
    print(len(soc) - 1)
    t -= 1

File: Squares_and_Cubes_1619B.cpp
#include <iostream>
#include <cmath>
#include <vector>
typedef long long int ll;
using namespace std;
ll soc(ll n)
{
    vector<ll>vec;
    ll r = sqrt(n) + 1;
    for(ll i = 1; i<=r; i++)
    {
        if (i*i <= n)
            vec.push_back(i*i);
        if(i*i*i <= n)
            vec.push_back(i*i*i);
    }
    sort(vec.begin(), vec.end());
    vec.erase(std::unique(vec.begin(), vec.end()), vec.end());
    return vec.size();
}
int main()
{
    ll t,n;
    cin>>t;
    while(t--)
    {
        cin>>n;
        cout<<soc(n)<<endl;
    }
    return 0;
}
